chain/src/txhashset/txhashset.rs

```diff
@@ -1496,19 +1496,29 @@ fn check_and_remove_files(txhashset_path: &PathBuf, header: &BlockHeader) -> Res
 			.difference(&pmmr_files_expected)
 			.cloned()
 			.collect();
+		let mut removed = 0;
 		if !difference.is_empty() {
-			debug!(
-				"Unexpected file(s) found in txhashset subfolder {:?}, removing.",
-				&subdirectory_path
-			);
-			for diff in difference {
+			for diff in &difference {
 				let diff_path = subdirectory_path.join(diff);
-				file::delete(diff_path.clone())?;
-				debug!(
-					"check_and_remove_files: unexpected file '{:?}' removed",
-					diff_path
-				);
+				if let Err(e) = file::delete(diff_path.clone()) {
```

Comment:
Since you handle both variants here I'd prefer to use` match`

Reply:
OK, will change it now:)